### PR TITLE
Fix scheduler initialization order

### PR DESCRIPTION
**Pull request by SunMarc**
# What does this PR do?

This PR put the `lr_scheduler` initialization after the optimizer is prepared to deal with any modification that might happen after. This is mainly to deal with deepspeed as the optimizer is modified and starting with torch 2.10, checks are more strict. 

Fixes https://github.com/huggingface/transformers/pull/43692

### CODE REVIEW

File: src/transformers/trainer.py [hunk 1]
```diff
@@ -2327,21 +2329,23 @@ def _inner_training_loop(
                 self._fsdp_qlora_plugin_updates()
                 if self.accelerator.mixed_precision != "fp8":
                     self.model = self.accelerator.prepare(self.model)
-            self.create_optimizer_and_scheduler(num_training_steps=max_steps)
```
Comment (qgallouedec) [inline src/transformers/trainer.py:2325]: wanna keep `create_optimizer_and_scheduler` for backward compatibility? Deprecating it is an option

**Review by qgallouedec — APPROVED** [review]
thanks, closing #43692

**Review by ArthurZucker — APPROVED** [review]
checks are stricter with 2.10, but this does not change anything for 2.4?

**Comment by SunMarc** [comment]
> checks are stricter with 2.10, but this does not change anything for 2.4?

Nope !

File: src/transformers/trainer.py (hunk 1 above)
Comment (SunMarc) [reply src/transformers/trainer.py:2325]: Yeah keeping it for BC for now but I might deprecate it slowly !
